### PR TITLE
change HEAL's aggMDS to use agg_server_paging

### PR DESCRIPTION
**Pull request by craigrbarnes**
Link to Jira ticket if there is one:

### Environments
qa-heal

### Description of changes
new config to match feat/agg_server_paging.

### CODE REVIEW

File: qa-heal.planx-pla.net/metadata/aggregate_config.json [hunk 1]
```diff
@@ -1,1240 +1,1296 @@
...
+          }
+        }
+      },
+      "commons_url": {},
+      "tags": {
+        "type": "array"
+      },
+      "advSearchFilters": {
+        "type": "array",
+        "default": []
```
Comment (aartivnkt) [inline qa-heal.planx-pla.net/metadata/aggregate_config.json:21]: @craigrbarnes what does this advsearch filters do, and why is it needed?

File: qa-heal.planx-pla.net/metadata/aggregate_config.json [hunk 2]
```diff
@@ -1,1240 +1,1296 @@
...
+      },
+      "study_description_summary": {},
+      "project_title": {}
+    },
+    "settings": {
+      "cache_drs": true
```
Comment (aartivnkt) [inline qa-heal.planx-pla.net/metadata/aggregate_config.json:27]: I thought everything in aggMDS is cached, why is there a separate cache_drs field?

File: qa-heal.planx-pla.net/metadata/aggregate_config.json [hunk 3]
```diff
@@ -1,1240 +1,1296 @@
...
+    }
+  },
+  "adapter_commons": {
+    "HEAL": {
+      "mds_url": "https://qa-heal.planx-pla.net/",
+      "commons_url": "https://qa-heal.planx-pla.net/",
+      "adapter": "gen3",
+      "config" : {
+        "guid_type": "unregistered_discovery_metadata",
```
Comment (aartivnkt) [inline qa-heal.planx-pla.net/metadata/aggregate_config.json:36]: is this "unregistered_discovery_metadata" only for unregistered HEAL studies? how about registered heal studies? I think that is discovery_metadata? how are those loaded in the aggMDS if guid_type is only refined for unregistered_discovery_metadata here?

File: qa-heal.planx-pla.net/metadata/aggregate_config.json [hunk 4]
```diff
@@ -1,1240 +1,1296 @@
...
+    },
+    "SAMHDA": {
+      "mds_url": "https://qa-heal.planx-pla.net/",
+      "commons_url": "https://qa-heal.planx-pla.net",
+      "adapter": "gen3",
+      "keep_original_fields": false,
+      "select_field": {
+        "field_name": "data_resource",
+        "field_value": "SAMHDA"
+      },
```
Comment (aartivnkt) [inline qa-heal.planx-pla.net/metadata/aggregate_config.json:63]: what does the select_field do?

Comment (aartivnkt) [reply qa-heal.planx-pla.net/metadata/aggregate_config.json:63]: Tested this new config in my dev env, it errors out not sure what I'm missing happy to sync later:
```
[2022-08-23 19:43:02,435][       mds][   INFO] Received 835 from HEAL
Traceback (most recent call last):
  File "/src/src/mds/populate.py", line 187, in <module>
    asyncio.run(main(commons_config=commons, hostname=args.hostname, port=args.port))
  File "/usr/local/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/src/src/mds/populate.py", line 135, in main
    await populate_metadata(name, common, results)
  File "/src/src/mds/populate.py", line 84, in populate_metadata
    await datastore.update_metadata(
  File "/src/src/mds/agg_mds/datastore/__init__.py", line 32, in update_metadata
    await client.update_metadata(*args)
  File "/src/src/mds/agg_mds/datastore/elasticsearch_dao.py", line 134, in update_metadata
    elastic_search_client.index(
  File "/env/lib/python3.9/site-packages/elasticsearch/client/utils.py", line 101, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/env/lib/python3.9/site-packages/elasticsearch/client/__init__.py", line 370, in index
    return self.transport.perform_request(
  File "/env/lib/python3.9/site-packages/elasticsearch/transport.py", line 402, in perform_request
    status, headers_response, data = connection.perform_request(
  File "/env/lib/python3.9/site-packages/elasticsearch/connection/http_urllib3.py", line 252, in perform_request
    self._raise_error(response.status, raw_data)
  File "/env/lib/python3.9/site-packages/elasticsearch/connection/base.py", line 253, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(
elasticsearch.exceptions.RequestError: RequestError(400, 'mapper_parsing_exception', 'object mapping for [tags] tried to parse field [tags] as object, but found a concrete value')
command terminated with exit code 1
```

**Comment by PlanXCyborg** [comment]
# qa-heal.planx-pla.net/manifest.json
## :warning: Services are being downgraded
- metadata

**Comment by PlanXCyborg** [comment]
# qa-heal.planx-pla.net/manifest.json
## :warning: Services are being downgraded
- metadata

**Comment by PlanXCyborg** [comment]
# qa-heal.planx-pla.net/manifest.json
## :warning: Services are being downgraded
- metadata

**Comment by PlanXCyborg** [comment]
# qa-heal.planx-pla.net/manifest.json
## :warning: Services are being downgraded
- metadata

File: qa-heal.planx-pla.net/metadata/aggregate_config.json [hunk 5]
```diff
@@ -23,9 +23,6 @@
                 "type": "array",
                 "default": []
             },
-            "data_dictionary": {
```
Comment (mfshao) [inline qa-heal.planx-pla.net/metadata/aggregate_config.json]: this field is causing issues for the new AggMDS schema feature, the error was `bytes can be at most 32766 in length`

one of the JCOIN metadata has a super long pure text data dict https://jcoin.datacommons.io/mds/metadata/1U2CDA050098-01_b. This is causing the issue. I think we should use text typle for these long fields, so maybe we should add another field type in the new AggMDS @craigrbarnes @tianj7

**Comment by PlanXCyborg** [comment]
# qa-heal.planx-pla.net/manifest.json
## :warning: Services are being downgraded
- metadata

**Comment by PlanXCyborg** [comment]
# qa-heal.planx-pla.net/manifest.json
## :warning: Services are being downgraded
- metadata

**Comment by PlanXCyborg** [comment]
# qa-heal.planx-pla.net/manifest.json
## Deployment changes
- metadata
  - configuration files will need to be updated (uc-cdis/metadata-service#65)
  - Database migration for MDS (uc-cdis/metadata-service#63)
## Breaking changes
- metadata
  - The configuration file is not compatible with the previous release. This is due to the new support for defining a schema, new configuration options, and changes to the adapters. (uc-cdis/metadata-service#65)

**Review by mfshao — APPROVED** [review]

**Pull request merged**